controller/src/beerocks/master/tasks/dynamic_channel_selection_task.h

```diff
@@ -0,0 +1,118 @@
+                                   task_pool &tasks_, std::string radio_mac_);
+    virtual ~dynamic_channel_selection_task() {}
+
+    typedef struct {
+        sMacAddr radio_mac;
+    } sTriggerSingleScan_event;
+
+    typedef struct {
+        sMacAddr radio_mac;
+    } sScanTriggered_event;
+
+    typedef struct {
+        sMacAddr radio_mac;
+    } sScanResultsReady_event;
+
+    typedef struct {
+        sMacAddr radio_mac;
+        sChannelScanResultsElement scan_results;
+    } sScanResultsDump_event;
+
+    typedef struct {
+        sMacAddr radio_mac;
+    } sScanFinished_event;
+
+    typedef struct {
+        sMacAddr radio_mac;
+    } sScanAbort_event;
+
+    typedef struct {
+        sMacAddr radio_mac;
+    } sScanEnableChange_event;
+
+#define FOREACH_DCS_EVENT(EVENT)                                                                   \
+    EVENT(INVALID_EVENT)                                                                           \
+    EVENT(TRIGGER_SINGLE_SCAN)                                                                     \
+    EVENT(SCAN_TRIGGERED)                                                                          \
+    EVENT(SCAN_RESULTS_READY)                                                                      \
+    EVENT(SCAN_RESULTS_DUMP)                                                                       \
+    EVENT(SCAN_FINISHED)                                                                           \
+    EVENT(SCAN_ABORTED)                                                                            \
+    EVENT(SCAN_ENABLE_CHANGE)
+
+    // Event ENUM
+    enum class eEvent { FOREACH_DCS_EVENT(GENERATE_ENUM) };
+
+protected:
+    virtual void work() override;
+    virtual void handle_event(int event_type, void *obj) override;
+    virtual void handle_events_timeout(std::multiset<int> pending_events) override;
+
+private:
+    void dcs_wait_for_event(eEvent cs_event);
+
+#define FOREACH_DCS_STATE(STATE)                                                                   \
+    STATE(INIT)                                                                                    \
+    STATE(IDLE)                                                                                    \
+    STATE(TRIGGER_SCAN)                                                                            \
+    STATE(WAIT_FOR_SCAN_TRIGGERED)                                                                 \
+    STATE(WAIT_FOR_RESULTS_READY)                                                                  \
+    STATE(WAIT_FOR_RESULTS_DUMP)                                                                   \
+    STATE(SCAN_DONE)                                                                               \
+    STATE(ABORT_SCAN)
+
+    // State ENUM
+    enum class eState { FOREACH_DCS_STATE(GENERATE_ENUM) };
+
+    // Strings Array
+    static const char *s_ar_states[];
+    static const char *s_ar_events[];
```

Comment:
Why not simply static arrays in the cpp file?

Reply:
removed dependence from fsm_move_state() .
moved to cpp.